### PR TITLE
fix zeroed_width issues, make default sampling midpoint

### LINKED ISSUE #3313: problems with `zeroed_width` argument in `pulse.continuous`
There seems to be some issues with initial/final discontinuity in some pulses in the pulse library.

```python
g_disc = discrete.gaussian(duration=240, amp=0.1, sigma=80)
g_disc.draw(scaling=0.2)
```
image

```python
g_disc = discrete.gaussian_square(duration=240, amp=0.1, sigma=15, risefall=20)
g_disc.draw(scaling=0.2)
```
image

### PR DESCRIPTION
**Pull request by ajavadia**
Fixes #3313 to correctly offset the pulses for Gaussian, Gaussian Square, Sech

Also makes default sampling midpoint

image

image

### CODE REVIEW

File: qiskit/pulse/pulse_lib/continuous.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def _fix_gaussian_width(gaussian_samples, amp: float, center: float, sigma: floa
     if zeroed_width is None:
         zeroed_width = 2*(center+1)
 
-    zero_offset = gaussian(np.array([-zeroed_width/2]), amp, center, sigma)
+    zero_offset = gaussian(np.array([zeroed_width]), amp, center, sigma)
```
[the remark this answers is not in the document]
Comment (lcapelluto) [reply qiskit/pulse/pulse_lib/continuous.py]: Why does `center` become 0?

**Comment by ajavadia** [comment]
Ok now the first/last samples are zeroed. Most of the updates are to docstrings.

I renamed `width` to `square_width` in `continuous.gaussian_square` because it's just the width of the square part.

File: qiskit/pulse/pulse_lib/continuous.py [hunk 2]
```diff
@@ -108,23 +109,22 @@ def _fix_gaussian_width(gaussian_samples, amp: float, center: float, sigma: floa
                         ret_scale_factor: bool = False) -> np.ndarray:
     r"""Enforce that the supplied gaussian pulse is zeroed at a specific width.
 
-    This is achieved by subtracting $\Omega_g(center \pm zeroed_width/2)$ from all samples.
+    This is achieved by subtracting $\Omega_g(center \pm zeroed_width)$ from all samples.
```
Comment (taalexander) [inline qiskit/pulse/pulse_lib/continuous.py]:
```suggestion
    This is achieved by subtracting $\Omega_g(center \pm zeroed_width/2)$ from all samples.
```

File: qiskit/pulse/pulse_lib/discrete.py [hunk 3]
```diff
@@ -167,7 +167,8 @@ def sin(duration: int, amp: complex, freq: float = None,
 def gaussian(duration: int, amp: complex, sigma: float, name: Optional[str] = None) -> SamplePulse:
```
Comment (taalexander) [inline qiskit/pulse/pulse_lib/discrete.py]: Not blocking, but should we expose the zeroing of the pulse as an option?

**Pull request merged**